lib/spack/spack/cmd/install.py

```diff
@@ -402,9 +401,14 @@ def get_tests(specs):
     # 2. Concrete specs from yaml files
     for file in args.specfiles:
         with open(file, 'r') as f:
-            s = spack.spec.Spec.from_yaml(f)
+            if file.endswith('json'):
+                s = spack.spec.Spec.from_json(f)
+            elif file.endswith('yaml'):
+                s = spack.spec.Spec.from_yaml(f)
```

Comment:
We should invert this and leave off the elif -- we previously haven't required a particular extension for these files.

```suggestion
            if file.endswith('yaml') or file.endswith('yml'):
                s = spack.spec.Spec.from_yaml(f)
            else:
                s = spack.spec.Spec.from_json(f)
```

Reply:
That makes sense. Initially I was going to throw an exception here if there was no match, but it messed up the entire structure...